src/components/ConnectToXeroButton/index.native.tsx

```diff
@@ -0,0 +1,101 @@
+import FullScreenLoadingIndicator from '@components/FullscreenLoadingIndicator';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import Modal from '@components/Modal';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import {removePolicyConnection} from '@libs/actions/connections';
+import getXeroSetupLink from '@libs/actions/connections/ConnectToXero';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {Session} from '@src/types/onyx';
+import type {ConnectToXeroButtonProps} from './types';
+
+type ConnectToXeroButtonOnyxProps = {
+    /** Session info for the currently logged in user. */
+    session: OnyxEntry<Session>;
+};
+
+function ConnectToXeroButton({policyID, session, shouldDisconnectIntegrationBeforeConnecting, integrationToDisconnect}: ConnectToXeroButtonProps & ConnectToXeroButtonOnyxProps) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+    const webViewRef = useRef<WebView>(null);
+    const [isWebViewOpen, setWebViewOpen] = useState(false);
+
+    const authToken = session?.authToken ?? null;
+
+    const renderLoading = () => <FullScreenLoadingIndicator />;
+    const [isDisconnectModalOpen, setIsDisconnectModalOpen] = useState(false);
+
+    return (
+        <>
+            <Button
+                onPress={() => {
+                    if (shouldDisconnectIntegrationBeforeConnecting && integrationToDisconnect) {
+                        setIsDisconnectModalOpen(true);
+                        return;
+                    }
+                    setWebViewOpen(true);
+                }}
+                text={translate('workspace.accounting.setup')}
+                style={styles.justifyContentCenter}
+                small
+            />
+            {shouldDisconnectIntegrationBeforeConnecting && isDisconnectModalOpen && integrationToDisconnect && (
+                <ConfirmModal
+                    title={translate('workspace.accounting.disconnectTitle', CONST.POLICY.CONNECTIONS.NAME.QBO)}
+                    onConfirm={() => {
+                        removePolicyConnection(policyID, integrationToDisconnect);
+                        setIsDisconnectModalOpen(false);
+                        setWebViewOpen(true);
+                    }}
+                    isVisible
+                    onCancel={() => setIsDisconnectModalOpen(false)}
+                    prompt={translate('workspace.accounting.disconnectPrompt', CONST.POLICY.CONNECTIONS.NAME.XERO)}
+                    confirmText={translate('workspace.accounting.disconnect')}
+                    cancelText={translate('common.cancel')}
+                    danger
+                />
+            )}
+            <Modal
+                onClose={() => setWebViewOpen(false)}
+                fullscreen
+                isVisible={isWebViewOpen}
+                type={CONST.MODAL.MODAL_TYPE.CENTERED}
```

Comment:
This modal type has a bug for android. More details here: https://github.com/Expensify/App/issues/40978